### ISSUE TITLE
Cannot use iter_mut() on mutable CxxVector?

### ISSUE THREAD

**Issue by NiceneNerd** [dtolnay/cxx#625]
As far as I can tell, if you acquire a mutable vector from C++ as `Pin<&mut CxxVector<T>>`, it's counterintuitively not possible to use `iter_mut()` with it. For example, in my bridge:
```rust
fn GetArray(self: Pin<&mut Byml>) -> Pin<&mut CxxVector<Byml>>;
```
The result when I try to iterate using `byml.GetArray().iter_mut()`:
```rust
error[E0596]: cannot borrow data in a dereference of `Pin<&mut CxxVector<ffi::Byml>>` as mutable
  --> src/byml/mod.rs:55:17
   |
55 |                 byml.GetArray()
   |                 ^^^^^^^^^^^^^^^ cannot borrow as mutable
   |
   = help: trait `DerefMut` is required to modify through a dereference, but it is not implemented for `Pin<&mut CxxVector<ffi::Byml>>````

Just to check, I also found it happens with more trivial cases like `Pin<&mut CxxVector<u8>>`.

**Comment by dtolnay** [comment]
Good catch, that method was accidentally uncallable. I published 1.0.25 with a fix to make it work on Pin\<\&mut CxxVector\>.

**Comment by NiceneNerd** [comment]
Awesome, thanks for the quick fix
